x-pack/plugins/security_solution/public/common/components/event_details/get_alert_summary_rows.tsx

```diff
@@ -17,88 +17,195 @@ import {
-  { id: 'destination.address' },
-  { id: 'destination.port' },
-  { id: 'process.name' },
-];
+      // If no primary category matches or hasn't been defined on purpose (e.g. in order to follow the source event)
+      // resolve more fields based on the other event categories.
+      if (allEventCategories?.includes(EventCategory.FILE)) {
+        fields = fields.concat(getFieldsByCategory({ primaryEventCategory: EventCategory.FILE }));
+      }
 
-const memoryShellCodeAlertFields: EventSummaryField[] = [
-  ...defaultDisplayFields,
-  { id: 'rule.name', label: ALERTS_HEADERS_RULE_NAME },
-  {
-    id: 'Target.process.thread.Ext.start_address_details.memory_pe.imphash',
-    label: ALERTS_HEADERS_TARGET_IMPORT_HASH,
-  },
-];
+      if (allEventCategories?.includes(EventCategory.PROCESS)) {
+        fields = fields.concat(
+          getFieldsByCategory({ primaryEventCategory: EventCategory.PROCESS })
+        );
+      }
+      return fields;
+  }
+}
 
-const behaviorAlertFields: EventSummaryField[] = [
-  ...defaultDisplayFields,
-  { id: 'rule.description', label: ALERTS_HEADERS_RULE_DESCRIPTION },
-];
+/**
+ * Gets the fields to display based on the event's code.
+ * Contains some enhancements to resolve more fields based on the event's categories.
+ * @param eventCode The event's code
+ * @param eventCategories The events categories
+ * @returns A list of fields to include
+ */
+function getFieldsByEventCode(
+  eventCode: string | undefined,
+  eventCategories: EventCategories
+): EventSummaryField[] {
+  switch (eventCode) {
+    case EventCode.BEHAVIOR:
+      return [
+        { id: 'rule.description', label: ALERTS_HEADERS_RULE_DESCRIPTION },
+        // Resolve more fields based on the source event
+        ...getFieldsByCategory({ ...eventCategories, primaryEventCategory: undefined }),
+      ];
+    case EventCode.SHELLCODE_THREAD:
+      return [
+        { id: 'Target.process.executable' },
+        {
+          id: 'Target.process.thread.Ext.start_address_detaiuls.memory_pe.imphash',
+          label: ALERTS_HEADERS_TARGET_IMPORT_HASH,
+        },
+        {
+          id: 'Memory_protection.unique_key_v1',
+        },
+      ];
+    case EventCode.MEMORY_SIGNATURE:
+      return [{ id: 'process.Ext.memory_region.malware_signature.all_names' }];
+    case EventCode.RANSOMWARE:
+      // Resolve more fields based on the source event
+      return getFieldsByCategory({ ...eventCategories, primaryEventCategory: undefined });
+    default:
+      return [];
+  }
+}
 
-const memorySignatureAlertFields: EventSummaryField[] = [
-  ...defaultDisplayFields,
-  { id: 'rule.name', label: ALERTS_HEADERS_RULE_NAME },
-];
+/**
+ * Returns a list of fields based on the event's rule type
+ */
+function getFieldsByRuleType(ruleType?: string): EventSummaryField[] {
+  switch (ruleType) {
+    case 'threshold':
+      return [
+        { id: 'kibana.alert.threshold_result.count', label: ALERTS_HEADERS_THRESHOLD_COUNT },
```

Comment:
@madirey or @marshallmain  - @janmonschke is updating the document summary section in the alert fly-out for 8.1, and was wondering if he should also add checks for `signal.*` fields as well here to account for users older data or will we be covered by the aad mappings?

Reply:
It looks like we are here: https://github.com/elastic/kibana/blob/481fb8f536880e09a73c5e35253b7bec8eacb22f/x-pack/plugins/security_solution/server/lib/detection_engine/routes/index/signal_aad_mapping.json#L77 , but not sure if I'm missing something